api/src/messaging/consumers.py

```diff
@@ -51,151 +60,32 @@ async def receive(self, text_data=None, byte_data=None):
-            "id": message_object.id,
-            "message_body": message_body,
-            "creator": creator,
-            "previous_message": previous_message,
-            "datetime_created": message_object.datetime_created.strftime(
-                "%Y-%m-%dT%H:%M:%S.%fZ"
-            ),
-            "tagged_users": tagged_users,
-            "content_type": text_data_json["content_type"],
-            "object_id": text_data_json["object_id"],
-        }
-
     # Sends a received message to the group
     async def send_like(self, event):
-        message_id = event["message_id"]
-        is_liked = event["like"]
-        liked_user_id = event["user_id"]
-        user_id = self.scope["user"].id
-
-        if self.member["id"] == user_id:
-            await self.send(
-                text_data=json.dumps(
-                    {
-                        "message_id": message_id,
-                        "like": is_liked,
-                        "user_id": liked_user_id,
-                    }
-                )
-            )
-
-        else:
-            await self.channel_layer.group_discard(
-                self.room_group_name, self.channel_name
-            )
+        data = {
+            "message_id": event["message_id"],
+            "like": event["like"],
+            "user_id": event["user_id"],
+        }
+        await self.send_data(data)
 
     async def send_message(self, event):
-        message_id = event["id"]
-        message_body = event["message_body"]
-        creator = event["creator"]
-        datetime_created = event["datetime_created"]
-        previous_message = event["previous_message"]
-        tagged_users = event["tagged_users"]
-        user_id = self.scope["user"].id
-        content_type = event["content_type"]
-        object_id = event["object_id"]
-
-        if self.member["id"] == user_id:
-            await self.send(
-                text_data=json.dumps(
-                    {
-                        "id": message_id,
-                        "message_body": message_body,
-                        "creator": creator,
-                        "previous_message": previous_message,
-                        "datetime_created": datetime_created,
-                        "tagged_users": tagged_users,
-                        "content_type": content_type,
-                        "object_id": object_id,
-                    }
-                )
-            )
+        data = {
+            "id": event["id"],
+            "message_body": event["message_body"],
+            "creator": event["creator"],
+            "previous_message": event["previous_message"],
+            "datetime_created": event["datetime_created"],
+            "tagged_users": event["tagged_users"],
+            "content_type": event["content_type"],
+            "object_id": event["object_id"],
+        }
+        await self.send_data(data)
 
+    # Helper function for creating data
+    async def send_data(self, data):
```

Comment:
Two things:
1. It doesn't create data, it sends data, specifically sending an event (hence the name, send_data). You should update the comments
2. Add type hints to this method
`async def send_data(self, data: dict) -> None:`